tests/privileged-tests/ui_get_sysname_ioctl_override.cpp

```diff
@@ -91,14 +93,16 @@ bool request_is_ui_get_sysname(unsigned long int request)
            static_cast<unsigned long>(UI_GET_SYSNAME(0));
 }
 
+template<typename Param1>
+auto request_param_type(int (* ioctl)(int, Param1, ...)) -> Param1;
 }
 
-extern "C" int ioctl(int fd, unsigned long int request, ...) __THROW
+extern "C" int ioctl(int fd, decltype(request_param_type(&ioctl)) request, ...) noexcept
```

Comment:
This is weird, but I guess `&ioctl` is still binding to the ioctl declaration in `linux/ioctl.h` at this point rather than being an infinitely recursive definition…

Reply:
Would you prefer this?
```
using ioctl_request_t = decltype(request_param_type(&ioctl));
extern "C" int ioctl(int fd, ioctl_request_t request, ...) noexcept
```